nixos/modules/services/web-apps/slamdata.nix

```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.slamdata;
+  stateDir = "/var/lib/quasar";
+  confFile0 = pkgs.writeText "quasar-config.json" (builtins.toJSON cfg.config);
+  confFile = "${stateDir}/config.json";
+  cmd = "${pkgs.jre}/bin/java -jar ${pkgs.slamdata}/quasar.jar"
+      + " -c ${confFile}"
+      + " -C ${pkgs.slamdata}/public";
+in
+
+{
+  options.services.slamdata = {
+      enable = mkEnableOption "SlamData";
+
+      config = mkOption {
+        type = types.attrs;
+        description = "Quasar configuration";
+        default = {
+          server.port = 8080;
+          mountings = { };
+        };
+      };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.slamdata = {
+      description = "SlamData";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      preStart = ''
+        mkdir -p "${stateDir}"
+        [ -f "${confFile}" ] || cp "${confFile0}" "${confFile}"
```

Comment:
Unfortunately it does. As you keep using SlamData and adding new workspaces, quasar saves the changes into the config file. I tried to go with the immutable config but quasar loudly complains on a first attempt to modify it.